### PR TITLE
Add implementation to get component names from Input manifest

### PR DESCRIPTION
**Pull request by Divyaasm**
### Description
Get component names from input manifest 

### Issues Related
https://github.com/opensearch-project/opensearch-build/issues/4005

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/jenkins/InputManifest.groovy [hunk 1]
```diff
@@ -53,12 +53,35 @@ class InputManifest {
...
+        String name
+        String repository
+
+        Component(Map data) {
+            this.name = data.name
+            this.repository = data.repository
+        }
+
+    }
+
```
Comment (peterzhuamazon) [inline src/jenkins/InputManifest.groovy:76]: Any reason not implementation the entire map like here?

https://github.com/opensearch-project/opensearch-build-libraries/blob/14bc0937325533f5541c53c9abfeac80b1e2c7df/src/jenkins/BundleManifest.groovy#L54-L72

Comment (Divyaasm) [reply src/jenkins/InputManifest.groovy:76]: Currently `Components` Property of input Manifest is not used any where, so I have only added arguments which are used by the release-branch creation[ job](https://github.com/opensearch-project/opensearch-build/pull/4520)

Comment (peterzhuamazon) [reply src/jenkins/InputManifest.groovy:76]: I do not agree with that approach.
If input manifest have those properties that you can use, just include them.
Bundle/Build manifest already have those why does input manifest be any different here? Unless it does not support those properties which is another case.

Comment (Divyaasm) [reply src/jenkins/InputManifest.groovy:76]: Sure I can do that. Thanks Peter

**Review by peterzhuamazon — APPROVED** [review]

**Pull request merged**